src/builder/commands/generic.rs

```diff
@@ -77,9 +97,9 @@ fn find_cmd<P:AsRef<Path>>(ctx: &Context, cmd: P)
         "-- empty PATH --".to_string()));
 }
 
-pub fn run_command_at_env(ctx: &mut Context, cmdline: &[String],
+fn build_command(ctx: &mut Context, cmdline: &[String],
```

Comment:
All this old stuff is structured like this from pre-`unshare` times, when we had no `Command` structure.

I mean you can make a function that makes some common settings and returns command. But you shouldn't add the arguments that are already there on current interface: `cmdline`, `env`, `path` (latter is essentially a workdir). This would make code nicer and even avoid some memory allocations.